src/webapp/src/auth/google.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@nestjs/common'
+import { SettingsService } from '../settings/settings.service'
+import { OAuth2Client } from 'google-auth-library'
+import { ConfigService } from '@nestjs/config'
+
+@Injectable()
+export class GoogleOAuth2Service {
+  constructor (
+    private readonly settingsService: SettingsService,
+    private readonly configService: ConfigService
+  ) {}
+
+  async getUserPayload (idToken: string): Promise<any> {
+    const keys = await this.settingsService.getKeysSettings()
+    const audience = (keys?.auth_google_client_id) ?? this.configService.get<string>('GOOGLE_CLIENT_ID')
+    const payload = await (await this._verify(idToken, audience)).getPayload()
+    return payload
```

Comment:
We should prob convert Google's payload into something with a well defined type.